akka-stream/src/main/scala/akka/stream/scaladsl/One2OneBidiFlow.scala

```diff
@@ -77,10 +75,7 @@ object One2OneBidiFlow {
 
       setHandler(outOut, new OutHandler {
         override def onPull(): Unit = pull(outIn)
-        override def onDownstreamFinish(): Unit = {
-          cancel(outIn)
-          cancel(inIn) // short-cut to speed up cleanup of upstream
-        }
+        override def onDownstreamFinish(): Unit = cancel(outIn)
```

Comment:
> point 4) above is not true, it will signal UnexpectedOutputException in this case

Yes, true. Appears to be a left-over from an earlier implementation.
I'll simply remove point 4) from the scaladoc.

> pullsSuppressed can only ever be 0 or 1, i.e. it should be a boolean

Yes. I always keep forgetting that there can only every be one pull and one push pending.

> GraphStageLogic.complete(...) should get the same treatment of Emitting states that completeStage() has, I suspect that this may be the cause of some problems you see

I don't understand what you mean by that. Can you clarify?